x-pack/filebeat/input/awss3/sqs_s3_event.go

```diff
@@ -185,6 +188,17 @@ func (p *sqsS3EventProcessor) keepalive(ctx context.Context, log *logp.Logger, w
 }
 
 func (p *sqsS3EventProcessor) getS3Notifications(body string) ([]s3EventV2, error) {
+	// Check if a parsing script is defined. If so, it takes precedence over
+	// format autodetection.
+	if p.script != nil {
+		var raw common.MapStr
+		dec := json.NewDecoder(strings.NewReader(body))
```

Comment:
I think SQS messages can take the form of XML and unstructured text (double check that). Maybe we should pass the raw string through and do the JSON decode in the `parse` implementation if it's required.

Reply:
Good point 👍